### PR TITLE
fix(dot/network): move low reputation peer removal from network ConnManager to peer scoring logic (dot/peerstate)

### PR DESCRIPTION
**Pull request by kishansagathiya**
## Changes

- moves low reputation peer removal from network ConnManager to peer scoring logic (dot/peerstate)
- After studying the code I realized that peer scoring logic was already taking care of not going above max peer count. But it was not working as expected.
- peerset service (doWorks goroutine) sends messages to network service
about peers to connect, drop, reject, disconnect etc. Some of those
message were being send without a peer id and set id. This was the
reason why our excess peers were not getting removed.
- To save ourselves from such a problem in future, processMessage now
checks if peer id is empty or not.
- Also change the resultMsgCh type to peerset.Message from interface{}

## Tests

```
./bin/gossamer --chain polkadot
```
check block import logs to see number of peers. Default number of max peer is 50. Expected behaviour is that, any point peers should not go above 50. Mostly it would not increase after 45. 
## Issues

- #2039 

## Primary Reviewer

- @noot

### CODE REVIEW

File: dot/peerset/peerset.go [hunk 1]
```diff
@@ -183,6 +183,8 @@ func NewConfigSet(in, out uint32, reservedOnly bool, allocTime time.Duration) *C
 	}
 
 	return &ConfigSet{
+		// Why are we using an array of config in the set, when we are
+		// using just one config
```
Comment (noot) [inline dot/peerset/peerset.go:187]: #1886 @arijitAD could you give more context on this?

if I remember correctly it was so we could score the different stream protocols differently (eg. sync, block-announce, etc)

File: dot/peerset/peerset.go [hunk 2]
```diff
@@ -546,11 +553,24 @@ func (ps *PeerSet) incoming(setID int, peers ...peer.ID) error {
...
 		case state.tryAcceptIncoming(setID, pid) != nil:
-			ps.resultMsgCh <- Message{Status: Reject}
+			ps.resultMsgCh <- Message{
+				Status: Reject,
+				setID:  uint64(setID),
+				PeerID: pid,
+			}
 		default:
-			ps.resultMsgCh <- Message{Status: Accept}
+			logger.Infof("incoming connection accepted from peer %s", pid)
```
Comment (noot) [inline dot/peerset/peerset.go:568]:
```suggestion
			logger.Debugf("incoming connection accepted from peer %s", pid)
```

**Review by noot — APPROVED** [review]
works for me! great work, just update the log level please :p

File: dot/peerset/peerset.go (hunk 2 above)
Comment (kishansagathiya) [reply dot/peerset/peerset.go]: addressed

File: dot/network/service.go [hunk 3]
```diff
@@ -670,6 +669,9 @@ func (s *Service) startPeerSetHandler() {
 
 func (s *Service) processMessage(msg peerset.Message) {
 	peerID := msg.PeerID
+	if peerID == "" {
+		logger.Errorf("found empty peer id in peerset message")
```
Comment (qdm12) [inline dot/network/service.go:673]: should we return here?

**Review by EclesioMeloJunior — APPROVED** [review]

File: dot/network/service.go (hunk 3 above)
Comment (kishansagathiya) [reply dot/network/service.go:673]: No, processMessage is used to execute network tasks. If peer id is empty, that means we cannot execute that particular task. We should continue to process other tasks. If we get an empty message, that mean we have written wrong code (somewhere we are sending mesage without peerid).

This check is a precautionary measure.

Comment (qdm12) [reply dot/network/service.go:673]: > If peer id is empty, that means we cannot execute that particular task. We should continue to process other tasks.

What do you mean by 'particular task'? In my view, `processMessage(msg)` is that particular task, and we should return if the peer id is empty right?

Comment (kishansagathiya) [reply dot/network/service.go:673]: Oh yes, my bad! Thanks for spotting this! Also, Fixed!

**Pull request merged**